runtime/src/snapshot_utils/archive_format.rs

```diff
@@ -36,8 +39,9 @@ impl TryFrom<&str> for ArchiveFormat {
             TAR_BZIP2_EXTENSION => Ok(ArchiveFormat::TarBzip2),
             TAR_GZIP_EXTENSION => Ok(ArchiveFormat::TarGzip),
             TAR_ZSTD_EXTENSION => Ok(ArchiveFormat::TarZstd),
+            TAR_LZ4_EXTENSION => Ok(ArchiveFormat::TarLz4),
             TAR_EXTENSION => Ok(ArchiveFormat::Tar),
-            _ => Err(ParseError::InvalidExtension),
+            _ => Err(ParseError(extension.to_string())),
```

Comment:
Introducing a new snapshot compression also means there's a period of time when nodes in our cluster do not recognize the new format.

Thanks for also improving the error message.  Too bad that the existing nodes running older version will still simply complain InvalidExtension without further information :sweat_smile:.